internal/gcsx/random_reader_stretchr_test.go

```diff
@@ -805,47 +805,59 @@ func (t *RandomReaderStretchrTest) Test_ReadFromMultiRangeReader_ReadChunk() {
 	}
 }
 
-func (t *RandomReaderStretchrTest) Test_ReadFromMultiRangeReader_ValidateTimeout() {
+func (t *RandomReaderStretchrTest) Test_ReadFromMultiRangeReader_ErrorHandling() {
 	testCases := []struct {
-		name      string
-		dataSize  int
-		sleepTime time.Duration
+		name               string
+		dataSize           int
+		sleepTime          time.Duration
+		expectedErrKeyword string
 	}{
 		{
-			name:      "TimeoutPlusOneMilliSecond",
-			dataSize:  100,
-			sleepTime: TestTimeoutForMultiRangeRead + time.Millisecond,
+			name:               "NilMRDWrapper",
+			dataSize:           100,
+			sleepTime:          0,
+			expectedErrKeyword: "MultiRangeDownloaderWrapper",
 		},
-		// Ensure that this is always the last test
 		{
-			name:      "TimeoutValue",
-			dataSize:  100,
-			sleepTime: TestTimeoutForMultiRangeRead,
+			name:               "TimeoutPlusOneMilliSecond",
+			dataSize:           100,
+			sleepTime:          TestTimeoutForMultiRangeRead + time.Millisecond,
+			expectedErrKeyword: "Timeout",
+		},
+		{
+			name:               "TimeoutValue",
+			dataSize:           100,
+			sleepTime:          TestTimeoutForMultiRangeRead,
+			expectedErrKeyword: "Timeout",
 		},
 	}
 
-	for i, tc := range testCases {
+	for _, tc := range testCases {
 		t.Run(tc.name, func() {
 			t.rr.wrapped.reader = nil
 			t.rr.wrapped.isMRDInUse = false
 			t.object.Size = uint64(tc.dataSize)
 			testContent := testutil.GenerateRandomBytes(int(t.object.Size))
-			fakeMRDWrapper, err := NewMultiRangeDownloaderWrapperWithClock(t.mockBucket, t.object, &clock.FakeClock{})
-			assert.Nil(t.T(), err, "Error in creating MRDWrapper")
-			t.rr.wrapped.mrdWrapper = &fakeMRDWrapper
-			t.mockBucket.On("NewMultiRangeDownloader", mock.Anything, mock.Anything).Return(fake.NewFakeMultiRangeDownloaderWithSleep(t.object, testContent, tc.sleepTime)).Times(1)
-			t.mockBucket.On("BucketType", mock.Anything).Return(gcs.BucketType{Zonal: true}).Times(1)
+			if tc.name == "NilMRDWrapper" {
```

Comment:
Its preferred to extract testcases in seperate methods if you need if-else. It becomes difficult to maintain and extend in future.
For eg, NilMRDWrapper doesn't require lot of this code. can you move it as seperate test method..